src/components/AutoTagBuilder/AutoTagBuilder.types.part.tsx

```diff
@@ -0,0 +1,52 @@
+import { LabeledInputProps, InputChangeEvent } from '../../common';
+
+export interface AutoTagBuilderAutosuggestItem {
+  key: string;
+  content?: React.ReactChild;
+}
+
+export interface AutoTagBuilderProps<T> extends LabeledInputProps<Array<T>> {
```

Comment:
you can introduce a new common type and use it both by boths `Autocomplete` and `AutocompleteTagBuilder` components

Reply:
These two types use a different approach for value handling, that means that Autocomplete has dedicated events for suggestions while AutocompleteTagBuilder uses suggestions as value field. Therefore, from my point, these types not so similar to have their own base class.